server/coordinator/shard_picker_test.go

```diff
@@ -19,25 +19,19 @@ func TestRandomShardPicker(t *testing.T) {
 	snapshot := c.GetMetadata().GetClusterSnapshot()
 
 	shardPicker := coordinator.NewRandomBalancedShardPicker()
-	shardNodes, err := shardPicker.PickShards(ctx, snapshot, 2, false)
-	re.NoError(err)
-
-	// Verify the number of shards and ensure that they are not on the same node.
-	re.Equal(len(shardNodes), 2)
-	re.NotEqual(shardNodes[0].NodeName, shardNodes[1].NodeName)
-
-	// ExpectShardNum is bigger than node number and enableDuplicateNode is false, it should be throw error.
-	_, err = shardPicker.PickShards(ctx, snapshot, 3, false)
-	re.Error(err)
 
 	// ExpectShardNum is bigger than node number and enableDuplicateNode is true, it should return correct shards.
-	shardNodes, err = shardPicker.PickShards(ctx, snapshot, 3, true)
+	shardNodes, err := shardPicker.PickShards(ctx, snapshot, 3)
 	re.NoError(err)
 	re.Equal(len(shardNodes), 3)
-	shardNodes, err = shardPicker.PickShards(ctx, snapshot, 4, true)
+	shardNodes, err = shardPicker.PickShards(ctx, snapshot, 4)
 	re.NoError(err)
 	re.Equal(len(shardNodes), 4)
 	// ExpectShardNum is bigger than shard number.
-	_, err = shardPicker.PickShards(ctx, snapshot, 5, true)
+	shardNodes, err = shardPicker.PickShards(ctx, snapshot, 5)
+	re.NoError(err)
+	re.Equal(len(shardNodes), 5)
+	shardNodes, err = shardPicker.PickShards(ctx, snapshot, 9)
 	re.NoError(err)
+	re.Equal(len(shardNodes), 9)
```

Comment:
Maybe we need add a check for shardNodes, we need to ensure that the shardNodes is average on nodes and shards.

Reply:
Add `TODO`.